lstchain/irf/hdu_table.py

```diff
@@ -309,15 +307,19 @@ def create_event_list(
 
     src_sky_pos = SkyCoord(
         alt=reco_alt, az=reco_az, frame=AltAz(obstime=time_utc, location=location)
-    ).transform_to(frame="icrs")
+    )
     tel_pnt_sky_pos = SkyCoord(
-        alt=pointing_alt.mean(),
-        az=pointing_az.mean(),
-        frame=AltAz(obstime=mean_time, location=location),
-    ).transform_to(frame="icrs")
+        alt=pointing_alt[0],
+        az=pointing_az[0],
+        frame=AltAz(obstime=time_utc[0], location=location),
+    )
+
+    with erfa_astrom.set(ErfaAstromInterpolator(30 * u.s)):
+        src_sky_pos_icrs = src_sky_pos.transform_to(frame="icrs")
+        tel_pnt_sky_pos_icrs = tel_pnt_sky_pos.transform_to(frame="icrs")
 
     # Observation modes
-    source_pointing_diff = source_pos.separation(tel_pnt_sky_pos)
+    source_pointing_diff = source_pos.separation(tel_pnt_sky_pos_icrs)
```

Comment:
This should also be `source_pos_icrs`, otherwise astropy will make the same transform twice

Reply:
The `source_pos` is already in ICRS as I pass it from either getting the source positions from the user input or from resolving the source name

Reply:
But you have a a transform_to('icrs') just above

Reply:
That is for `src_sky_pos` which is coming from the reco position of each observed event.
Maybe, the variable name is confusing? I can change the `source_pos` to `source_std_pos` maybe?

Reply:
No, use reco_pos or similar for the reconstructed positiom AMD Reserve source for the assumed / observed source.

Reply:
Let me propose `reco_altaz` and `reco_icrs`

Reply:
And accordingly, for pointing position, `pnt_altaz` and `pnt_icrs`?